src/net/sourceforge/kolmafia/webui/RelayServer.java

```diff
@@ -96,16 +96,28 @@ public static final void stop() {
   @Override
   public void run() {
     boolean startedSuccessfully = true;
-    Integer minPort = 60080;
-    Integer maxPort = minPort + 10;
+
+    int minPort, maxPort;
+    int relayPort = Preferences.getInteger("relayPort");
```

Comment:
could you add it to defaults, please?

Reply:
Ooh, I thought there must be something like this :) With which value would you add it? 0 or 60080?
If I add it with 60080, it will not work together with the "try 10 ports by default, but not if one has been set explicitly".

Reply:
I've added it as 0 for now.

Reply:
I don't think this will be successful if one character has no relayPort set and another has relayPort set to 60080 and that character logs in second.   I'd consider renaming the preference to something like characterSpecificRelayPort and having some error checking that flagged a potential error if a specific port was in the range of ports that would be checked in the absence of a specific port.  Some kind of test might be appropriate if anyone is clever enough to figure it out.

Reply:
I think this is fine. Change that error message to tell the user that their explicitly chosen port is already in use if that is the case.

Reply:
I agree. I think the cases "relayPort is set" and "relayPort is not set" are distinct. The efforts to reuse the existing `minPort` / `maxPort` stuff I think is more confusing.

I would like to see:
* is relayPort nonzero? If so, start the server there, else print error message about relayPort being inaccessible.
* otherwise, do what we did originally with a port range, printing the error message about minPort and maxPort and informing about `relayPort`.

Reply:
Yes, this is also how I would have implemented your suggestions. I will do it like this tonight and get back to you.

Reply:
I've implemented the changes as suggested.